pkgs/tools/networking/openssh/common.nix

```diff
@@ -96,6 +96,10 @@ stdenv.mkDerivation {
 
   ${if stdenv.hostPlatform.isStatic then "NIX_LDFLAGS" else null}= [ "-laudit" ] ++ lib.optionals withKerberos [ "-lkeyutils" ];
 
+  # openssh calls passwd to allow the user to reset an expired password, but nixos
+  # doesn't ship it at /usr/bin/passwd.
+  PATH_PASSWD_PROG = "/run/wrappers/bin/passwd";
```

Comment:
This should probably be only set when `stdenv.isLinux` (even if it is unlikely that macOS users are going to run nixpkgs sshd :) )

Reply:
Ah, I was wondering about that. Is there any way to determine that we're on nixos specifically, not just Linux? Again - unlikely that a non-nixos Linux user would be running openssh; but...

Reply:
Unforunately no, there is also a patch in glibc iirc that just goes with "every linux is nixos"

Reply:
Most straightforward way I can think of would be adding an overridable arg to the top of the file, e.g. `isNixOS ? true`

Reply:
How about `isNixOS ? false` and then override it to true in `nixos/modules/programs/ssh.nix`?

Reply:
Then it wouldn't be cached by hydra. IMO the priority should be to cache the version of openssh used by NixOS users.

Reply:
yeah, fair enough